tidy3d/components/structure.py

```diff
@@ -248,34 +248,25 @@ def get_monitor_name(index: int, data_type: str) -> str:
         return monitor_name_map[data_type]
 
     def make_adjoint_monitors(
-        self, freqs: list[float], index: int
+        self, freqs: list[float], index: int, field_keys: list[str]
```

Comment:
ah, right. I think the fix is to pass the `sim_fields_keys` into the `make_adjoint_monitors`. Since you have the optimizer, that would mean getting

```py
sim = optimizer.design.simulation
traced_fields_sim = setup_run(simulation=sim)
sim_fields_keys = list(traced_fields_sim.keys())
```

Reply:
I don't think `setup_run` will work if this is not called inside a `grad` call?

Reply:
not sure, guess you could just do

```

    return simulation.strip_traced_fields(
        include_untraced_data_arrays=False, starting_path=("structures",)
    )

```
which is what setup run does.

Reply:
But the point is that there are no traced fields since it's not a simulation called in `grad` no?

Reply:
Basically what we're trying to do there is get the gradient monitors from the static definition of the InverseDesign object.

Reply:
ah ok. yea I guess we dont know. but we can effectively "fake" it then, because we know it's a Box with a CustomMedium and we need the `permittivity` data.

So I think one approach could just be to pass
```
field_keys=(f'structures/{(len(sim.structures) - 1}/medium/permittivity', )
```
into the `make_adjoint_monitors` call. Or you can just make the monitors manually since again we know it's just a box so it needs a similarly sized field and permittivity monitor

Reply:
Using the first approach, works for now, obviously should be dealt with better in the future..